### PR TITLE
build pahole from source

### PR DESCRIPTION
**Pull request by brycekahle**
To avoid [a bug](https://bugs.launchpad.net/ubuntu/+source/dwarves/+bug/2025370) with Ubuntu's `pahole`, and to allow us to use newer flags.

### CODE REVIEW

**Comment by brycekahle** [comment]
FYI I'm not able to test this Action myself, but it *should* work. If you have a way to test this change before merge, then please do.

**Review by rafaeldtinoco — APPROVED** [review]
LGTM, lets see if this works or I can try to do something else similar way.

**Comment by rafaeldtinoco** [comment]
I have reverted this one as it did not work. Will get something else similar to this to have latest pahole always built.

**Pull request merged**
